### PR TITLE
8299478: [Lilliput/JDK17] Fix Windows build

### PR DESCRIPTION
**Pull request by rkennke**
These changes should fix the Windows (and MacOS) build of Lilliput/JDK17. I'm going blindly, here, only guided by GHA.

Ok?

Thanks,
Roman

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8299478](https://bugs.openjdk.org/browse/JDK-8299478): [Lilliput/JDK17] Fix Windows build

### Reviewers
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/lilliput-jdk17u pull/1/head:pull/1` \
`$ git checkout pull/1`

Update a local copy of the PR: \
`$ git checkout pull/1` \
`$ git pull https://git.openjdk.org/lilliput-jdk17u pull/1/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1`

View PR using the GUI difftool: \
`$ git pr show -t 1`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/lilliput-jdk17u/pull/1.diff">https://git.openjdk.org/lilliput-jdk17u/pull/1.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/oops/objArrayOop.hpp [hunk 1]
```diff
@@ -77,7 +77,7 @@ class objArrayOopDesc : public arrayOopDesc {
     size_t size_words = align_up(base_offset_in_bytes() + asz, HeapWordSize) / HeapWordSize;
     size_t osz = align_object_size(size_words);
     assert(osz < max_jint, "no overflow");
-    return osz;
+    return static_cast<int>(osz);
```
Comment (shipilev) [inline src/hotspot/share/oops/objArrayOop.hpp:80]: `checked_cast`?

**Review by shipilev — COMMENTED** [review]
Can you explain a bit why this should be `int`, and not `size_t`, like the rest of upstream `object_size()` methods?

**Comment by rkennke** [comment]
> Can you explain a bit why this should be `int`, and not `size_t`, like the rest of upstream `object_size()` methods?

It should be size_t, and it is size_t in tip, but it is int in jdk17, and I did not want to go down the rabbit hole of changing this and all the dependencies, and thereby risk to introduce funny bugs. The code is made specifically to not overflow int, and it even checks that. I believe we should be good.

**Review by shipilev — APPROVED** [review]
All right, let's see what it breaks.

**Comment by rkennke** [comment]
Thanks!
/integrate

**Pull request closed without merging**